azurelinuxagent/ga/exthandlers.py

```diff
@@ -38,44 +38,52 @@
 import azurelinuxagent.common.version as version
 from azurelinuxagent.common.cgroupconfigurator import CGroupConfigurator
 from azurelinuxagent.common.datacontract import get_properties, set_properties
-from azurelinuxagent.common.errorstate import ErrorState, ERROR_STATE_DELTA_INSTALL
-from azurelinuxagent.common.event import add_event, WALAEventOperation, elapsed_milliseconds, report_event
-from azurelinuxagent.common.exception import ExtensionError, ProtocolError, ProtocolNotFoundError, \
-    ExtensionDownloadError, ExtensionErrorCodes, ExtensionUpdateError, ExtensionOperationError
+from azurelinuxagent.common.errorstate import ERROR_STATE_DELTA_INSTALL, ErrorState
+from azurelinuxagent.common.event import add_event, elapsed_milliseconds, report_event, WALAEventOperation, add_periodic
+from azurelinuxagent.common.exception import ExtensionDownloadError, ExtensionError, ExtensionErrorCodes, \
+    ExtensionOperationError, ExtensionUpdateError, ProtocolError, ProtocolNotFoundError, ExtensionStatusError
 from azurelinuxagent.common.future import ustr
-from azurelinuxagent.common.protocol.restapi import ExtHandlerStatus, \
-    ExtensionStatus, \
-    ExtensionSubStatus, \
-    VMStatus, ExtHandler
+from azurelinuxagent.common.protocol.restapi import ExtensionStatus, ExtensionSubStatus, ExtHandler, ExtHandlerStatus, \
+    VMStatus
+from azurelinuxagent.common.types import AttributeEnum
 from azurelinuxagent.common.utils.flexible_version import FlexibleVersion
-from azurelinuxagent.common.version import AGENT_NAME, CURRENT_VERSION, GOAL_STATE_AGENT_VERSION, \
-    DISTRO_NAME, DISTRO_VERSION, PY_VERSION_MAJOR, PY_VERSION_MINOR, PY_VERSION_MICRO
+from azurelinuxagent.common.version import AGENT_NAME, CURRENT_VERSION, DISTRO_NAME, DISTRO_VERSION, \
+    GOAL_STATE_AGENT_VERSION, PY_VERSION_MAJOR, PY_VERSION_MICRO, PY_VERSION_MINOR
+
+
+_HANDLER_PATTERN = r'^([^-]+)-(\d+(?:\.\d+)*)'
+_HANDLER_PKG_PATTERN = re.compile(_HANDLER_PATTERN + r'\.zip$', re.IGNORECASE)
+_DEFAULT_EXT_TIMEOUT_MINUTES = 90
 
 # HandlerEnvironment.json schema version
-HANDLER_ENVIRONMENT_VERSION = 1.0
+_HANDLER_ENVIRONMENT_VERSION = 1.0
 
-EXTENSION_STATUS_ERROR = 'error'
-EXTENSION_STATUS_SUCCESS = 'success'
-VALID_EXTENSION_STATUS = ['transitioning', 'error', 'success', 'warning']
-EXTENSION_TERMINAL_STATUSES = ['error', 'success']
+VALID_EXTENSION_STATUS = AttributeEnum(('transitioning', 'warning', 'error', 'success'))
```

Comment:
seems like this could be private as well

Reply:
Done.